xrootd/protocol/protocol.go

```diff
@@ -102,3 +106,97 @@ const (
 	// DataServer indicates whether this is a data server.
 	DataServer
 )
+
+// EntryStat holds the entry name and the entry stat information.
+type EntryStat struct {
+	EntryName     string // EntryName is the name of entry.
+	HasStatInfo   bool   // HasStatInfo indicates if the following stat information is valid.
+	ID            int64  // ID is the OS-dependent identifier assigned to this entry.
+	EntrySize     int64  // EntrySize is the decimal size of the entry.
+	IsExecutable  bool   // IsExecutable indicates that entry is either an executable file or a searchable directory.
+	IsDirectory   bool   // IsDirectory indicates that entry is a directory.
+	IsOther       bool   // IsOther indicates that entry is neither a file nor a directory.
+	IsOffline     bool   // IsOffline indicates that the file is not online (i. e., on disk).
+	IsReadable    bool   // IsReadable indicates that read access to that entry is allowed.
+	IsWritable    bool   // IsWritable indicates that write access to that entry is allowed.
+	IsPoscPending bool   // IsPoscPending indicates that the file was created with kXR_posc and has not yet been successfully closed.
```

Comment:
the meaning of a file opened with `kXR_posc` could be perhaps explained here (or on the method, if that field is converted into a method)